### ISSUE TITLE
Amazing project!

### ISSUE THREAD

**Issue by blackrosezy** [cazala/synaptic#3]
Great job @cazala, ¡Genial!. I like the way you provides the examples, very interesting scenarios especially _Paint an image_.

**Comment by EdwardGonzalez** [comment]
@cazala hey! :) how do I save my current status as JSON/XML/CSV whatever works, this has a lot of potential, so, I'd like the trained status in the server so that my clients won't have to execute it again, and my clients are given a ready to use neural network, I plan to use it to teach children and build my AI based teacher. :3

Saludos desde Venezuela!

**Comment by cazala** [comment]
@EdwardGonzalez the Network class has a `toJSON()` [method](https://github.com/cazala/synaptic/wiki/Networks#tojsonfromjson), so you can simply store that, and then you can recreate the network by using the static method `Network.fromJSON()` that will return a new instance which is a copy from the original network.

The weights are updated every time you backpropagate the error (from output layer to input layer) ie. every time you call `.propagate()`. During the training the propagate method is called on each iteration, so yes, the weights are changed at the end of every iteration.

Saludos desde Argentina (;

**Comment by EdwardGonzalez** [comment]
@cazala so you thought about everything, well today is time to make some tests, I'll also will try to create some network listener of some sort, because we can just consider the following.

I'll give you feedback, however I'm not pretty sure about using this issue as a way to do that.

**Comment by cazala** [comment]
sounds good! feel free to create new issues, i ll be glad to help as much as i can :)

**Comment by Nicomalacho** [comment]
@cazala i have been using synaptic, i really like it, but how i get the weights of a perceptron ?, i need it to print all the info of the network.

**Comment by cazala** [comment]
@Nicomalacho you can use myNetwork.toJSON() to get all the parameters

**Comment by Nicomalacho** [comment]
@cazala yes i allready did it, but if i want to print a specific weight conection ?.

**Comment by cazala** [comment]
@Nicomalacho ok, then you can call myNetwork.restore() to populate all the values from the optimized array buffer to the actual objects, and then navigate to the specific layer, neuron, connection and weight, for example if you want the weight of the first projected connection of the first neuron of the input layer: 

myNetwork.layers.input.list[0].projected[0].weight

**Comment by Nicomalacho** [comment]
thanks!!, i solved it, using myNetwork.toJSON(), and then with the json, ["connections"]["0"]["weight"].

**Comment by cazala** [comment]
awesome (;

**Comment by Nicomalacho** [comment]
@cazala hi again, im working in a project at my university and im going to make a evolutionary algorithm, so i need to set the weights in a neural network. so how can i do that? by the way congrats, really nice project. 
Saludos desde Colombia.

**Comment by cazala** [comment]
Check out [what this guy did](https://github.com/ivanseidel/IAMDinosaur), it may shed some light on how to use synaptic + genetic algorithms
